backend/app/api/endpoints/share.py

```
@@ -0,0 +1,491 @@
+    Used by share link page to display KB info and handle permission requests.
+
+    - **resource_id**: Knowledge base ID
+    """
+    try:
+        return knowledge_share_service.get_kb_share_info(
+            db=db,
+            knowledge_base_id=resource_id,
+            user_id=current_user.id,
+        )
+    except ValueError as e:
+        raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)